### PR TITLE
Fix type mismatch issue in sumMapFiltered

### LINKED ISSUE #57616: sumMapFiltered aggregate function does not work with Int64
I run these two queries and I expect the same result:

```sql
SELECT sumMapFiltered([1,2,3])(a,b) FROM values('a Array(UInt64), b Array(UInt64)',([1, 2, 3], [10, 10, 10]), ([3, 4, 5], [10, 10, 10]),([4, 5, 6], [10, 10, 10]),([6, 7, 8], [10, 10, 10]));

SELECT sumMapFiltered([1,2,3])(a,b) FROM values('a Array(Int64), b Array(Int64)',([1, 2, 3], [10, 10, 10]), ([3, 4, 5], [10, 10, 10]),([4, 5, 6], [10, 10, 10]),([6, 7, 8], [10, 10, 10]));
```
Output in latest version:
```
([1,2,3],[10,10,20])
([],[])
```

It works for UInt64 correctly. It does not work for Int64.

### PR DESCRIPTION
**Pull request by Chen768959**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
fixed #57616
This problem occurs because all positive number arguments are automatically identified as `UInt64` type, leading to an inability to match Int type data in `sumMapFiltered`. The issue of non-matching is indeed confusing, as the `UInt64` parameters are not specified by the user. 
Additionally, if the arguments are `[1,2,3,toInt8(-3)]`, due to the `getLeastSupertype()`, these parameters will be uniformly treated as `Int` type, causing `'1,2,3'` to also fail in matching the `UInt` type data in `sumMapFiltered`.

### CODE REVIEW

File: src/AggregateFunctions/AggregateFunctionSumMap.cpp [hunk 1]
```diff
@@ -545,7 +545,24 @@ class AggregateFunctionSumMapFiltered final :
         }
     }
 
-    bool keepKey(const Field & key) const { return keys_to_keep.contains(key); }
+    bool keepKey(const Field & key) const
+    {
+        if (keys_to_keep.contains(key))
+            return true;
+
+        if (key.getType() == Field::Types::UInt64)
```
Comment (alexey-milovidov) [inline src/AggregateFunctions/AggregateFunctionSumMap.cpp:553]: Comments are needed.

Comment (Chen768959) [reply src/AggregateFunctions/AggregateFunctionSumMap.cpp:553]: Okay, I will add some comments.

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 2ce673baaeb88f1cb75a2bbdf0a6091af3ef1708 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58408/2ce673baaeb88f1cb75a2bbdf0a6091af3ef1708/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58408/2ce673baaeb88f1cb75a2bbdf0a6091af3ef1708/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by jsc0218** [comment]
It is still relevant, @Chen768959 would you mind to continue? Looks like just use `safeGet` would be okay. I would review it.

**Pull request merged**